swan/scripts/others/userconfig.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# Author: Danilo Piparo, Enric Tejedor, Pedro Maximino, Diogo Castro 2023
+# Copyright CERN
+# Here the user session is configured, including the home directory and 
+# the terminal.
+
+_log () {
+    if [[ "$*" == "ERROR:"* ]] || [[ "$*" == "WARNING:"* ]] || [[ "${JUPYTER_DOCKER_STACKS_QUIET}" == "" ]]; then
+        echo "$@"
+    fi
+}
+
+START_TIME_SETUP_SWAN_HOME=$( date +%s.%N )
+SETUP_SWAN_HOME_TIME_SEC=$(echo $(date +%s.%N --date="$START_TIME_SETUP_SWAN_HOME seconds ago") | bc)
+_log "user: $USER, host: ${SERVER_HOSTNAME%%.*}, metric: configure_user_env_swan_home.duration_sec, value: $SETUP_SWAN_HOME_TIME_SEC"
+
+# FIXME: Move to extensions.
+# Persist enabled notebook nbextensions
+SWAN_PROJECTS=$HOME/SWAN_projects
+NBCONFIG=$JPY_DIR/nbconfig
+mkdir -p $NBCONFIG
+LOCAL_NB_NBEXTENSIONS=$SWAN_PROJECTS/.notebook_nbextensions
+if [ ! -f $LOCAL_NB_NBEXTENSIONS ]; then 
+  echo "{
+    \"load_extensions\": {
+    }
+  }" > $LOCAL_NB_NBEXTENSIONS
+fi
+rm -f $NBCONFIG/notebook.json
+ln -s $LOCAL_NB_NBEXTENSIONS $NBCONFIG/notebook.json
+
+START_TIME_SETUP_LCG=$( date +%s.%N )
+
+# Setup LCG
+source $LCG_VIEW/setup.sh
+
+SETUP_LCG_TIME_SEC=$(echo $(date +%s.%N --date="$START_TIME_SETUP_LCG seconds ago") | bc)
+_log "user: $USER, host: ${SERVER_HOSTNAME%%.*}, metric: configure_user_env_cvmfs.${ROOT_LCG_VIEW_NAME:-none}.duration_sec, value: $SETUP_LCG_TIME_SEC"
+
+# Add SWAN modules path to PYTHONPATH so that it picks them
+export PYTHONPATH=/usr/local/lib/swan/extensions/:$PYTHONPATH
+
+# Run user startup script
+export JUPYTER_DATA_DIR=$LCG_VIEW/share/jupyter 
```

Comment:
This export is probably not well placed? It's not related with the user startup script. I would do it for example after line 32.

I believe it's still needed to find notebook extensions in the LCG release -- please @PMax5 add it to the list of things to try.

Reply:
Moved below line 32!